src/library/starrating.cpp

```diff
@@ -30,20 +31,22 @@ StarRating::StarRating(
 }
 
 QSize StarRating::sizeHint() const {
-    return PaintingScaleFactor * QSize(m_maxStarCount, 1);
+    return m_scaleFactor * getbaseSize();
 }
 
-void StarRating::paint(QPainter *painter, const QRect &rect) const {
+void StarRating::paint(QPainter* painter, const QRect& rect, int height) {
+    m_scaleFactor = height > 0 ? height : kDefaultPaintingScaleFactor;
     // Assume the painter is configured with the right brush.
     painter->setRenderHint(QPainter::Antialiasing, true);
     painter->setPen(Qt::NoPen);
 
-    int yOffset = (rect.height() - PaintingScaleFactor) / 2;
+    // Center the painter baseline (y-origin of the polygons)
+    int yOffset = (rect.height() - m_scaleFactor) / 2;
     painter->translate(rect.x(), rect.y() + yOffset);
-    painter->scale(PaintingScaleFactor, PaintingScaleFactor);
+    painter->scale(m_scaleFactor, m_scaleFactor);
 
     //determine number of stars that are possible to paint
-    int n = rect.width() / PaintingScaleFactor;
+    int n = rect.width() / m_scaleFactor;
```

Comment:
`mixxx::TrackRecord::kMaxRating` from https://github.com/mixxxdj/mixxx/blob/b4a40b876cfeafd6b5f3926a8468d6a2e6cb40f4/src/track/trackrecord.h#L78

Reply:
`m_maxStarCount` is the paint limit, the painter only draws polygons (star or diamond) up to that number.

There's no limit for the rect, sure it can be wider than 5 stars. The StarEditor considers all mouse events in the rect, see `StarEditor::starAtPosition()`.